frontend/src/composables/use-sprite.ts

```diff
@@ -0,0 +1,42 @@
+import icons from "~/assets/svg/sprite/icons.svg"
+import images from "~/assets/svg/sprite/images.svg"
+import licenses from "~/assets/svg/sprite/licenses.svg"
+
+export const sprites = { icons, images, licenses }
+export type Sprite = keyof typeof sprites
+
+function isSprite(sprite: string): sprite is Sprite {
+  return Object.keys(sprites).includes(sprite)
+}
+
+function generateName(name: string) {
+  return name
+    .toLowerCase()
+    .replace(/\.svg$/, "")
+    .replace(/[^a-z0-9-:]/g, "-")
+    .replace(/:/g, "--")
+}
+
+export const useSprite = (name: string) => {
```

Comment:
Is this the module you said was vendored in? Where does it come from? Is it https://github.com/nuxt-modules/svg-sprite? We should include a source, credit, attribution, and link to an issue with a plan to remove the need for vendoring it.

Reply:
I added the comments to `use-sprite`, as well as `VIcon` and `VSvg`